packages/core/src/lib/GooglePlay.ts

```diff
@@ -146,24 +141,76 @@ export class GooglePlay {
    * @param packageName - The packageName of the versionCode we are looking up.
    */
   async getLargestVersionCode(packageName: string): Promise<number> {
-    const edit = await this._googlePlayApi.edits.insert({packageName: packageName});
-    const editId = edit.data.id;
-    if (!editId) {
-      throw new Error('Could not create a Google Play edit');
-    }
+    const editId = await this.startPlayOperation(packageName);
     const bundleResponse =
       await this._googlePlayApi.edits.bundles.list({packageName: packageName, editId: editId});
     if (!bundleResponse.data.bundles) {
       throw new Error('No bundles found from Google Play');
     }
     const versionCode = Math.max(
         ...bundleResponse.data.bundles.map((bundle) => bundle.versionCode!!));
-    // cleanup
-    await this._googlePlayApi.edits.delete({editId: editId, packageName: packageName});
+
+    await this.endPlayOperation(packageName, editId);
 
     return versionCode;
   }
 
+  /**
+   * Starts an edit on the Play servers. This is the basis for any play publishing api operation.
+   * @param packageName - the packageName of the app we want to interact with.
+   *
+   */
+  private async startPlayOperation(packageName: string): Promise<string> {
+    const edit = await this._googlePlayApi.edits.insert({packageName: packageName});
+    const editId = edit.data.id;
+    if (!editId) {
+      throw new Error('Could not create a Google Play edit');
+    }
+    return editId;
+  }
+
+  /**
+   * Cancels the edit in progress on the Play server.
+   * @param packageName - The packageName of the app we are interacting with.
+   * @param editId - The editId that is currently in progress.
+   */
+  private async endPlayOperation(packageName: string, editId: string): Promise<void> {
```

Comment:
Do we need an `endPlayOperation` for every `startPlayOperation`? If so, we could create a method like `performPlayOperation` that takes a callback, calls `startPlayOperation`, executes the callback (passing in the `editId` and then calls `endPlayOperation`.

That way it's easier for people to not forget the `end` part.

Reply:
I don't think we want to delete the edit on edits that we make a commit call. The delete documentation isn't super clear on this : https://developers.google.com/android-publisher/edits

Reply:
@PEConn - I am a little confused on how to do this in typescript. Each method that I would want to call has a different return type (one is void, one is number, another is bool). I am also interested in knowing how I would append an editId to a callback function that is being passed in.

Reply:
I spoke with someone here, I think i got it.